providers/summary/clearlydefined.js

```diff
@@ -181,28 +189,44 @@ class ClearlyDescribedSummarizer {
     setIfValue(result, 'licensed.declared', license)
   }
 
-  addPodData(result, data) {
+  addPodData(result, data, coordinates) {
     setIfValue(result, 'described.releaseDate', extractDate(data.releaseDate))
     setIfValue(result, 'described.projectWebsite', get(data, 'registryData.homepage'))
     const license = get(data, 'registryData.license')
     if (license) {
       setIfValue(result, 'licensed.declared', SPDX.normalize(typeof license === 'string' ? license : license.type))
     }
+
+    const homepage = get(data, 'registryData.homepage')
+    const revision = get(data, 'registryData.sourceInfo.revision')
+
+    const registry = buildRegistryUrl(coordinates)
+    const version = homepage && revision ? `${homepage}/tree/${revision}` : null
+    const download = homepage && revision ? `${homepage}/archive/${revision}.zip` : null
+    setIfValue(result, 'described.urls', { registry, version, download })
   }
 
-  addGemData(result, data) {
+  addGemData(result, data, coordinates) {
     setIfValue(result, 'described.releaseDate', extractDate(data.releaseDate))
     const license = SPDX.normalize(get(data, 'registryData.license'))
     if (license) set(result, 'licensed.declared', license)
     else {
       const licenses = SPDX.normalize((get(data, 'registryData.licenses') || []).join(' OR '))
       setIfValue(result, 'licensed.declared', licenses)
     }
+    setIfValue(result, 'described.urls', buildUrls(coordinates))
   }
 
-  addPyPiData(result, data) {
+  addPyPiData(result, data, coordinates) {
     setIfValue(result, 'described.releaseDate', extractDate(data.releaseDate))
     setIfValue(result, 'licensed.declared', data.declaredLicense)
+    const releases = get(data, 'registryData.releases')
+    // TODO: we are currently picking the first url that contains a tar.gz extension
+    // we should understand what's the correct process on a pypi definition that contains multiple object for the same release
+    const revision = find(releases[coordinates.revision], revision => revision.filename.includes('tar.gz'))
+    const registry = buildRegistryUrl(coordinates)
+    const version = buildVersionUrl(coordinates)
+    setIfValue(result, 'described.urls', { registry, version, download: revision.url })
```

Comment:
setIfValue doesn't do much here since we are providing an object for the value inline. It will never be falsy and not set.

You may want to setIfValue each individual property

Reply:
Right, thanks